modules/post/windows/gather/enum_av_excluded.rb

```diff
@@ -0,0 +1,148 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core'
+require 'rex'
+
+class Metasploit3 < Msf::Post
+  include Msf::Post::Windows::Registry
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name'          => 'Windows Antivirus Excluded Locations Enumeration',
+        'Description'   => 'This module will enumerate all excluded directories within supported AV products',
+        'License'       => MSF_LICENSE,
+        'Author'        => [
+          'Andrew Smith', # original metasploit module
+          'Jon Hart <jon_hart[at]rapid7.com>' # improved metasploit module
+        ],
+        'Platform'      => [ 'win' ],
+        'SessionTypes'  => [ 'meterpreter' ]
+      )
+    )
+
+    register_options(
+      [
+        OptBool.new('DEFENDER', [true, 'Enumerate exclusions for Microsoft Defener', true]),
+        OptBool.new('ESSENTIALS', [true, 'Enumerate exclusions for Microsoft Security Essentials', true]),
+        OptBool.new('SEP', [true, 'Enumerate exclusions for Symantec Endpoint Protection (SEP)', true])
+      ]
+    )
+  end
+
+  def enum_mssec
+    if registry_enumkeys("HKLM\\SOFTWARE\\Microsoft").include?("Microsoft Antimalware")
+      print_status "MS Security Essentials Identified"
+      return true
+    else
+      return false
+    end
+  rescue
```

Comment:
My eye starts twitching with naked rescues. Can we please use `begin`? And also specify the class of exception you're expecting? The Pokemon pattern can lead to trouble. Thanks!

There are a few cases in the other `enum_*` functions too.